src/js/index.js

```diff
@@ -26,7 +26,7 @@ function renderBlogCategories(selector, containerId) {
   });
   const categories = Array.from(document.querySelectorAll(selector))
   categories.forEach(category => {
-    ReactDOM.render(React.createElement(BlogPostsContainer), category)
+    ReactDOM.render(<BlogPostsContainer card-data={category.dataset}/>, category)
```

Comment:
you can do `React.createElement(BlogPostsContainer, category.dataset)` and then you can avoid the `this.props['card-data']`